lib/core/src/Cardano/Pool/Metrics.hs

```diff
@@ -352,12 +355,44 @@ newStakePoolLayer tr db@DBLayer{..} nl metadataDir = StakePoolLayer
                     , apparentPerformance
                     , cost = poolCost
                     , margin = poolMargin
+                    , desirability = 0
                     }
                 , poolOwners
                 )
 
-    sortByPerformance :: [(StakePool, a)] -> [(StakePool, a)]
-    sortByPerformance = sortOn (Down . apparentPerformance . fst)
+        rankPools xs =
+            let
+                totalStake = sum $ map (fromIntegral . getQuantity . stake . fst) xs
+                relativeStake p = fromIntegral (getQuantity $ stake p) / totalStake
+
+                -- Does not matter. It only scales the resulting desirability.
+                memberStake = 0.01
+
+                toM (p, _) = P.Pool
+                        { P.leaderStake = 0
+                        , P.stake = relativeStake p
+                        , P.cost = fromIntegral $ getQuantity $ cost p
+                        , P.margin = fromIntegral (getPercentage $ margin p) / 100
+                        , P.recentAvgPerformance = apparentPerformance p
+                        }
+
+            in  map (\((pool, x), des) -> (pool {desirability = des}, x))
+                . sortOn (Down . snd)
+                . (P.nonMyopicMemberRewards jormungandrEnv toM memberStake)
+                $ xs
+
+    -- | Constant needed for ranking pools
+    jormungandrEnv :: P.Env
+    jormungandrEnv = P.Env
+        { P.leaderStakeInfluence = 0
+        , P.punishmentHarshness = 1 -- TODO: What should this be?
+        , P.optimalNumberOfPools = 100
+        , P.totalRewards = 1000000*1000000 -- TODO: What should this be?
```

Comment:
Will determine the influence of fixed `cost` on the ranking.

Seems non-trivial (but perhaps possible) to calculate from jörmungandr's reward constants. If e.g. tx fees can affect it significantly, it might be tricky to take into account.

I suspect it's okay-ish to ship with a hard-coded estimation. But pools with high `cost` will be ranked unfairly.

Reply:
*will try calculating from jörmungandr's config params*

Reply:
Ideally we wouldn't duplicate the calculation. Hopefully we can ask the Haskell Shelley Node for `R` over the LocalStateQuery protocol.

Reply:
Not needed in this PR. Will be tackled in #1276